dvc/remote/gdrive.py

```diff
@@ -324,10 +325,40 @@ def _create_remote_dir(self, parent_id, title):
 
     @gdrive_retry
     def _delete_remote_file(self, remote_id):
+        from pydrive2.files import ApiRequestError
+
         param = {"id": remote_id}
         # it does not create a file on the remote
         item = self.drive.CreateFile(param)
-        item.Delete()
+
+        try:
+            item.Trash() if self._trash_only else item.Delete()
+        except ApiRequestError as exc:
+            if (
+                not exc.error.get("code", None)
```

Comment:
feels like these two conditions are too complicated, especially this one with two negations - really hard to read and understand - could we simplify it all?